src/index.js

```diff
@@ -2,23 +2,46 @@
  * @flow
  */
 
-// example of generating HTML with js
+import store, {
+  getMessageSubject,
+  getMessageSender,
+  getMessageTimestamp,
+  getMessageSnippet,
+} from './store';
+
+function getThreadIds() {
+  let threadIds;
+  const { INBOX } = store.mailboxes;
+  if (INBOX) {
+    threadIds = INBOX.threadIds;
+  } else {
+    threadIds = [];
+  }
+  return threadIds;
+}
+
 function renderSidebar() {
   const sidebarContents = `
     <h2 class="email-header">Inbox</h2>
-    <ul class="email-list">
-      <li>
-        <button class="email-item" type="button">
-          <div class="sender-details">
-            <p>Email sender</p>
-            <span>Timestamp</span>
-          </div>
-          <p class="email-subject">Email1 subject</p>
-          <p>Email snippet</p>
-        </button>
-      </li>
-    </ul>
-  `;
+    <ul class="email-list">${getThreadIds()
+    .map((threadId) => {
+      const { messages } = store.threads[threadId];
+      const messageMetadata = messages[messages.length - 1];
+      const message = store.messages[messageMetadata.id];
+      return `<li>
+            <button class="email-item" type="button">
+              <div class="sender-details">
+                <p>${getMessageSender(message).split('\\')[0] || ''}</p>
```

Comment:
Nice way of grabbing the name! I would handle this in a separate little helper function though called something like `formatSender` or something so it's not cluttering up your template string here.